### PR TITLE
BaSD-MS-25: Update Sessions Logic

### PR DESCRIPTION
**Pull request by ravegliamartin**
## Brief
|           | Controller Method | Request Method and Path | Validations Used |
|:--------------:|------------|-----------------|:--------------------------------------|
| Create    | createSession()   | POST - /sessions/      | validations.validateSessions |
| Update    | updateSession()   | PUT - /sessions/:id    | validations.validateIdFormat validations.validateSessionUsedAttr |
| Delete    | deleteSession()   | DELETE - /sessions/:id | validations.validateIdFormat  |
| List All  | listAllSessions() | GET - /sessions/       | ---    |
| ~~List ById~~ | ~~listSession()~~     | ~~GET - /sessions/:id~~    | ~~validations.validateIdFormat~~    |
## Session Schema 

```
{
    postulant: {
      type: mongoose.Schema.Types.ObjectId,
      required: true,
      ref: 'Postulants',
    },
    psychologist: {
      type: mongoose.Schema.Types.ObjectId,
      required: true,
      ref: 'Psychologists',
    },
    status: {
      type: String,
      enum: ['cancelled', 'assigned', 'succesful'],
      default: 'assigned',
      required: true,
    },
    date: { type: Date, required: true },
    notes: { type: String },
},
```

### Minor Changes
- Postman Collection Updated.
- Deprecated `data/session.json`  deleted.

### CODE REVIEW

File: controllers/sessions.js [hunk 1]
```diff
@@ -1,71 +1,78 @@
...
 };
 
-const deleteSession = (req, res) => {
-  let removedSession;
-  const filteredSession = Sessions.filter((session) => {
-    if (session.id === req.query.id) {
-      removedSession = session;
-      return false;
+const listAllSessions = (req, res) => {
+  Sessions.find()
```
Comment (estefrare) [inline controllers/sessions.js]: add filters

File: controllers/sessions.js [hunk 2]
```diff
@@ -1,71 +1,78 @@
...
-  });
-  if (!updatedSession) res.status(404).send('Session NOT found');
-  fs.writeFile('./data/sessions.json', JSON.stringify(updatedSessions), {}, (error) => {
-    if (error) {
-      return res.status(400).send(error);
+    if (!deletedSession) {
+      return res.status(404).json({ msg: `The session 'id' (${req.params.id}) given  does not exist.` });
     }
-    return res.status(200).json(updatedSession);
+    return res.status(204).send(deletedSession);
```
Comment (estefrare) [inline controllers/sessions.js]: 204 not allow data response, just send a message

File: controllers/sessions.js [hunk 3]
```diff
@@ -1,71 +1,78 @@
...
+      date: req.body.date,
+      notes: req.body.notes,
+    },
+    { new: true },
+    (err, newSession) => {
+      if (err) {
+        return res.status(400).json(err);
+      }
+      if (!newSession) {
+        return res.status(404).json({ msg: `The session 'id' (${req.params.id}) given  does not exist.` });
```
Comment (estefrare) [inline controllers/sessions.js]: update the error format

File: validations/sessions.js [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+      res,
+    );
+  }
+  return next();
+};
+
+const validateSessionsUsedAttr = (req, res, next) => {
+  const invalidParams = [];
+  if (req.body.postulant && !ObjectId.isValid(req.body.postulant)) {
+    invalidParams.push("'Postulant ID'");
```
Comment (estefrare) [inline validations/sessions.js]: why are you using `"` and `'`?

File: validations/sessions.js [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+  }
+  if (invalidParams.length === 1) {
+    return errorResHelper(
+      `Param ${invalidParams[0]} is missing or invalid`,
+      res,
+    );
+  }
+  if (invalidParams.length > 1) {
+    return errorResHelper(
+      `Params ${invalidParams
```
Comment (estefrare) [inline validations/sessions.js]: I am not sure if `Params` is the correct word

**Review by estefrare — CHANGES_REQUESTED** [review]

File: controllers/sessions.js (hunk 2 above)
Comment (leimeter-joaquin) [reply controllers/sessions.js]: not even a message, just write 
```
return res.status(204).send()
```

File: validations/sessions.js (hunk 4 above)
Comment (ravegliamartin) [reply validations/sessions.js]: To add the single quotes to the string, it isn't the right way to do it?

**Review by ignacioaristo — APPROVED** [review]

**Review by estefrare — APPROVED** [review]

**Pull request merged**